File: test-case/char/chars1d/chars1d.pl
use strict;
use warnings;

# step [1]: implement solution function
# @param {string[]} chars
# @return {string[]}
sub double_list {
    my ($chars) = @_;
    my @arr;
    push @arr, @{$chars};
    push @arr, @{$chars};
    \@arr;
}

use Soda::Unittest::Utils qw(from_stdin);
use Soda::Unittest::TestWork qw(create);
# step [2]: setup function/return/arguments
my $work = create(\&double_list);
# $work->validator(sub { $_[0] vs $_[1]; return boolean });
$work->compare_serial(1);
print $work->run(from_stdin());

File: test-case/char/chars1d/chars1d.js
const {TestWork, Utils} = require('soda/unittest');

/**
 * @param {character[]} chars
 * @return {character[]}
 */
var doubleList = function(chars) {
    return chars.concat(chars);
};

const work = TestWork.create(doubleList);
work.compareSerial = true;
console.log(work.run(Utils.fromStdin()));

